apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-spring5/src/main/java/co/elastic/apm/agent/springwebflux/WebfluxServletHelper.java

```diff
@@ -60,19 +61,23 @@ public static Transaction<?> getServletTransaction(ServerWebExchange exchange) {
 
         // While the active transaction is the one created by Servlet, it would rely on the fact that we are on the
         // same thread as the one that created the transaction, which is an implementation detail.
-
-        Transaction<?> transaction = null;
-
         ServerHttpRequest exchangeRequest = exchange.getRequest();
+        return getTransactionFromRequest(exchangeRequest);
+    }
+
+    @Nullable
+    private static Transaction<?> getTransactionFromRequest(ServerHttpRequest exchangeRequest) {
         if (exchangeRequest instanceof AbstractServerHttpRequest) {
             Object nativeRequest = ((AbstractServerHttpRequest) exchangeRequest).getNativeRequest();
 
             // note: attribute name is defined in Servlet plugin and should be kept in sync
-            transaction = (Transaction<?>) getServletAttribute(nativeRequest, "co.elastic.apm.agent.servlet.ServletApiAdvice.transaction");
+            return (Transaction<?>) getServletAttribute(nativeRequest, "co.elastic.apm.agent.servlet.ServletApiAdvice.transaction");
 
+        } else if (exchangeRequest instanceof ServerHttpRequestDecorator) {
```

Comment:
Looks like this was a case we simply missed before. Starting with spring 3.4.0 the requests seem to be wrapped by some firewall by default.